components/schema-migrator/migrations/director/20221219120419_add-self-fa-and-drop-last-operation.up.sql

```diff
@@ -0,0 +1,19 @@
+BEGIN;
+
+ALTER TABLE formation_assignments
+    DROP COLUMN last_operation_initiator,
+    DROP COLUMN last_operation_initiator_type,
+    DROP COLUMN last_operation;
+
+INSERT INTO formation_assignments (id, formation_id, tenant_id, source, source_type, target, target_type, state)
+SELECT uuid_generate_v4(), fa.formation_id, fa.tenant_id, fa.source, fa.source_type, fa.source, fa.source_type, 'READY'
+FROM (
+         (
+             SELECT DISTINCT fa_source.formation_id, fa_source.tenant_id, fa_source.source, fa_source.source_type
+             FROM formation_assignments AS fa_source
+                      INNER JOIN formation_assignments AS fa_target
+                                 ON fa_source.source = fa_target.target AND fa_source.source_type = fa_target.target_type
```

Comment:
We should cross check the targets as well, otherwise imagine that we have the following formation - X, Y and Z.
X is being unassigned and everyone has async webhooks.

Then we have the following assignments: (X,Y), (Y,X), (X,Z), (Z,X), (Y,Z), (Z,Y)

Z has processed that X is being removed, however, X also processed that it is being removed from Y, however, Y still didn't processed that X is removed and X didn't process that is being removed from Z.

This will result in the following assignments:
(X,Y), (Z,X), (Y,Z), (Z,Y)

The first two will satisfy your condition and a (X,X) assignment will be created, which is not correct.

We should do full cross check `ON fa_source.source = fa_target.target AND fa_source.source_type = fa_target.target_type AND fa_source.target = fa_target.source AND fa_source.target_type = fa_target.source_type`